src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,78 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import React, { useState } from 'react';
+
+type Props = {
+  onStatusChange: (status: string) => void;
+  onSearchChange: (query: string) => void;
+  onClearSearch: () => void;
+  searchQuery: string;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onStatusChange,
+  onSearchChange,
+  onClearSearch,
+  searchQuery,
+}) => {
+  const [status, setStatus] = useState<string>('all');
```

Comment:
The `status` state is redundant because the parent component already manages the status filter. Consider removing this state and using the `status` prop directly from the parent component to avoid inconsistencies.